packages/vue-app/template/utils.js

```diff
@@ -258,7 +258,11 @@ export async function setContext (app, context) {
       }
     }
     if (process.server) {
-      app.context.beforeNuxtRender = fn => context.beforeRenderFns.push(fn)
+      app.context.beforeNuxtRender = fn => {
+        console.warn('`context.beforeNuxtRender(fn)` is deprecated in favour of `context.beforeRender(fn)`. The function updating the `nuxtState` has to be synchronous. Learn more on https://nuxtjs.org/docs/2.x/internals-glossary/context#beforenuxtrender')
+        context.beforeRenderFns.push(fn)
+      }
+      app.context.beforeRender = fn => context.beforeRenderHooks.push(fn)
```

Comment:
We are still during server rendering process. Using beforeSerialize or similar new name would probably be more clear (and less confusing between old beforeNuxtRender) 😊